### ISSUE TITLE
Cannot add parameter "typed_keys" to Search Application Search aggregation

### ISSUE THREAD

**Issue by l-trotta** [elastic/elasticsearch#106190]
### Elasticsearch Version

8.12.2

### Java Version

_bundled_

### OS Version

elasticsearch cloud

### Problem Description

When defining a SearchApplication template containing an aggregation, there is currently no way to add the `typed_keys` parameter defined for [Aggregations](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations.html#return-agg-type), or at least I didn't find a way to add it, nor did I manage to find documentation about it.

Support for `typed_keys`, in particular `typed_keys=true` is crucial for client libraries for strongly typed languages (e.g. Java, .Net, Go) as they allow identifying the concrete aggregation result class that should be instantiated.

### Steps to Reproduce

In Kibana:

```
PUT test-typed-keys-index
```

Example of the parameter used in a simple Aggregation
```
GET /test-typed-keys-index/_search?typed_keys
{
  "aggs": {
    "my-agg-name": {
      "histogram": {
        "field": "date",
        "interval": 1000
      }
    }
  }
}
```
With the result being
```
{
...
  "aggregations": {
    "histogram#my-agg-name": {
      "buckets": []
    }
  }
}
```

Creating a search application template using the same aggregation:
```
PUT _application/search_application/my_search_application
{
  "indices": ["test-typed-keys-index"],
  "template": {
    "script": {
      "source": """
        {
          "aggs": {
            "my-agg-name": {
              "histogram": {
                "field": "date",
                "interval": 1000
              }
            }
          }
        }
      """,
      "params": {
        "text_fields": [
          {"name": "date", "boost": 10}
        ],
        "explain": false,
        "from": 0,
        "size": 10
      }
    }
  }
}
```
Search Application Search: 

```
POST _application/search_application/my-app/_search?typed_keys
{
  "params": {
    "query_string": "*"
  }
}
```

Which fails with 

```
"type": "illegal_argument_exception",
"reason": "request [/_application/search_application/my-app/_search] contains unrecognized parameter: [typed_keys]"
```

**Comment by andreas-elastic** [comment]
Is there an update or a timeline for this issue? The bug prevents using a search application that uses aggregations and is called using a strongly typed language client. One of my customers has to add a `#` in front of their aggregation names as a workaround.

**Comment by liranabn** [comment]
The [PR](https://github.com/elastic/elasticsearch/pull/108007) was merged and will be available in the upcoming release.

**Comment by ioanatia** [comment]
going to close this as it is fixed with 8.14.0
